fdbclient/FileBackupAgent.actor.cpp

```diff
@@ -4102,6 +4102,8 @@ struct StartFullRestoreTaskFunc : RestoreTaskFuncBase {
 		}
 
 		state bool logsOnly = wait(restore.onlyAppyMutationLogs().getD(tr, false, false));
+		state bool inconsistentSnapshotOnly = wait(restore.inconsistentSnapshotOnly().getD(tr, false, false));
+		state Version firstConsistentVersion = invalidVersion;
 		if (beginVersion == invalidVersion) {
 			beginVersion = 0;
```

Comment:
I am not sure if this should be done.  In a mode where `logsOnly=false` and `inconsistentSnapshotOnly=true`, this will cause `beginVersion=0` which then becomes `firstVersion` and is used to initialize the mutations applier at that version.   I think this would make the selective range-only restore go unacceptably slow by causing the mutations applier to waste time reading all version ranges starting from 0, which would all be empty.  If this is correct, it could be one of the problems we saw recently.

Reply:
While this is existing code, I think we should figure it out while we are looking at this stuff.

Reply:
There will be a separate PR for the issues found here.